en/protocol/camera_def.md

```diff
@@ -0,0 +1,893 @@
+
+The Camera Definition File also provides support for localization where all strings to be translated are embedded within the file itself. The GCS will then select the appropriate language to use based on the availability of localized strings found within the file.
+
+## Schema
+
+The XML file has 3 main sections (elements):
+
+* Definition
+* Parameters
+* Localization
+
+### Definition
+
+All fields are self explanatory:
+
+```XML
+    <definition version="1">
+        <model>T100</model>
+        <vendor>Foo Industries</vendor>
+    </definition>
+```
+
+### Parameters
+
+Parameters can be simple or quite complex depending on the behavior they change. At a minimum, it defines the description (what is shown to the user) and the options. 
+
+#### Parameter Types
+
+The type of the parameter follows the enum [MAV_PARAM_EXT_TYPE](http://mavlink.org/messages/common#MAV_PARAM_EXT_TYPE_UINT8). Within the XML file, these are defined as:
+
+* bool (internally treated as a uint8)
+* uint8
+* int8
+* uint16
+* int16
+* uint32
+* int32
+* uint64
+* int64
+* float
+* double
+* custom
+
+The `custom` type is a special case. It allows for arbitrary data structures of up to 128 bytes but are not supported by default. You would need to extend or write your own camera controller within the GCS to interpret it.
+
+#### Parameter Definition
+
+The simplest parameter would be a boolean type, which inherently (and automatically) only provides two (on/off) options:
+
+```XML
+        <parameter name="CAM_IRLOCK" type="bool" default="0">
+            <description>Enable IR Lock</description>
+        </parameter>
+```
+
+The `name` attribute is the name of the parameter. It’s the name used when requesting or setting its value using the extended parameter messages.
+The `description` is what is shown to the user.
+
+More common are parameters that provide options:
+
+```XML
+        <parameter name="CAM_WBMODE" type="uint32" default="0">
+            <description>White Balance Mode</description>
+            <options>
+                <option name="Auto" value="0" />
+                <option name="Incandescent" value="1" />
+                <option name="Sunset" value="3" />
+                <option name="Sunny" value="4" />
+                <option name="Cloudy" value="5" />
+                <option name="Fluorescent" value="7" />
+            </options>
+        </parameter>
+```
+
+In this case, the GCS will automatically build a drop down list with the options defined within the `options` group. When sending/receiving the options, the `value` field is used and it is not in any way interpreted by the GCS. The `name` field is used for display only. In other words, using the example above, when the user selects *Sunset*, the GCS will send a [PARAM\_EXT\_SET](http://mavlink.org/messages/common#PARAM_EXT_SET) message with the id `CAM_WBMODE` and a uint32 value of 3.
```

Comment:
"When sending/receiving the options, the `value` field is used and it is not in any way interpreted by the GCS"

This confuses me - "the value field is used and it is not ... ". What is it used for?

Reply:
The intent is to say only the option *name* is shown to the user and when a selection is made the *value* is sent to the camera without any interpretation. In the example, the user selects *Sunset* and the GCS will send whatever value found in the `value` attribute of that option. It happens to be 3 in this example but could be whatever the camera defines.